core/templates/dev/head/collection_editor/CollectionEditorSavePublishDirective.js

```diff
@@ -0,0 +1,142 @@
+      'COLLECTION_EDITOR_INITIALIZED_COLLECTION',
+      'COLLECTION_EDITOR_UPDATED_COLLECTION',
+      'UNDO_REDO_SERVICE_CHANGE_APPLIED',
+      function(
+          $scope, $modal, alertsService, UndoRedoService,
+          CollectionEditorStateService, CollectionValidationService,
+          CollectionRightsBackendApiService,
+          WritableCollectionBackendApiService,
+          COLLECTION_EDITOR_INITIALIZED_COLLECTION,
+          COLLECTION_EDITOR_UPDATED_COLLECTION,
+          UNDO_REDO_SERVICE_CHANGE_APPLIED) {
+        $scope.collection = CollectionEditorStateService.getCollection();
+        $scope.isPrivate = GLOBALS.isPrivate;
+        $scope.canUnpublish = GLOBALS.canUnpublish;
+        $scope.validationIssues = [];
+        $scope.isSaveInProgress = (
+          CollectionEditorStateService.isSavingCollection);
+        var _collectionId = GLOBALS.collectionId;
+
+        var _validateCollection = function() {
+          if ($scope.isPrivate) {
+            $scope.validationIssues = (
+              CollectionValidationService
+                .findValidationIssuesForPrivateCollection(
+                  $scope.collection));
+          } else {
+            $scope.validationIssues = (
+              CollectionValidationService
+                .findValidationIssuesForPublicCollection(
+                  $scope.collection));
+          }
+        };
+
+        $scope.$on(
+          COLLECTION_EDITOR_INITIALIZED_COLLECTION, _validateCollection);
+        $scope.$on(COLLECTION_EDITOR_UPDATED_COLLECTION, _validateCollection);
+        $scope.$on(UNDO_REDO_SERVICE_CHANGE_APPLIED, _validateCollection);
+
+        $scope.getChangeListCount = function() {
+          return UndoRedoService.getChangeCount();
+        };
+
+        $scope.isCollectionSaveable = function() {
+          return (
+            $scope.getChangeListCount() > 0 &&
+            $scope.validationIssues.length === 0);
+        };
+
+        $scope.isCollectionPublishable = function() {
+          return (
+            $scope.isPrivate &&
+            $scope.getChangeListCount() === 0 &&
+            $scope.validationIssues.length === 0);
+        };
+
+        $scope.saveChanges = function() {
+          var isPrivate = $scope.isPrivate;
+          var modalInstance = $modal.open({
+            templateUrl: 'modals/saveCollection',
+            backdrop: true,
+            controller: [
+              '$scope', '$modalInstance', function($scope, $modalInstance) {
+                $scope.isCollectionPrivate = isPrivate;
+
+                $scope.save = function(commitMessage) {
+                  $modalInstance.close(commitMessage);
+                };
+                $scope.cancel = function() {
+                  $modalInstance.dismiss('cancel');
+                };
+              }
+            ]
+          });
+
+          modalInstance.result.then(function(commitMessage) {
+            // An explicit save is needed to push all changes to the backend at
```

Comment:
Not totally sure I understand this. What's the alternative to "an explicit save"?